Base/Python/slicer/util.py

```diff
@@ -2395,6 +2395,116 @@ def createProgressDialog(parent=None, value=0, maximum=100, labelText="", window
       setattr(progressIndicator, key, value)
   return progressIndicator
 
+def getPythonConsoleWidget():
+  import slicer
+  return slicer.util.mainWindow().pythonConsole().parent()
+
+from contextlib import contextmanager
+
+@contextmanager
+def peakPythonConsole(show=True):
```

Comment:
With this PR, we would now have the following functions:

| slicer.util API | equivalent API |
|---|---|
| slicer.util.pythonShell | slicer.app.pythonConsole |
| slicer.util.getPythonConsoleWidget | slicer.util.mainWindow().pythonConsole().parent() |

There are two issues with introducing the new function `getPythonConsoleWidget`:
* function is inconsistently named. Name `pythonShellWidget` would be consistent
* The suffix `Widget` is not always "correct". Indeed, when the settings `Python/DockableWindow` is false, the console is standalone dialog.

Considering this, I suggest to avoid adding the function `getPythonConsoleWidget` and rename `peakPythonConsole` to `raisePythonShell` and update the implementation to consider the value of `Python/DockableWindow` settings to conditionally get the parent widget.

For context:

https://github.com/Slicer/Slicer/blob/aaa9f918a4ee4f35bf09e05c5d7e53d083b2e4e3/Base/QTApp/qSlicerMainWindow.cxx#L456-L477

Reply:
Thanks, @jcfr. I think I have addressed this.